asf_public_discourse_home_decarbonisation/pipeline/data_processing_flows/flow_utils.py

```diff
@@ -0,0 +1,267 @@
+    """
+    Load data from S3 location.
+
+    Args:
+        bucket_name (str) : The S3 bucket name
+        file_path (str): S3 key to load
+    Returns:
+        Union[pd.DataFrame, Dict]: the loaded data
+    """
+    s3 = get_s3_resource()
+
+    obj = s3.Object(bucket_name, file_path)
+    if fnmatch(file_path, "*.json"):
+        file = obj.get()["Body"].read().decode()
+        return json.loads(file)
+    elif fnmatch(file_path, "*.csv"):
+        return pd.read_csv("s3://" + bucket_name + "/" + file_path)
+    elif fnmatch(file_path, "*.parquet"):
+        return pd.read_parquet("s3://" + bucket_name + "/" + file_path)
+    elif fnmatch(file_path, "*.pkl") or fnmatch(file_path, "*.pickle"):
+        file = obj.get()["Body"].read().decode()
+        return pickle.loads(file)
+    else:
+        logger.error(
+            'Function not supported for file type other than "*.csv", "*.json", "*.pkl" or "*.parquet"'
+        )
+
+
+def get_mse_category_data(category: str, collection_date: str) -> pd.DataFrame:
+    """
+    Gets data from a specific Money Saving Expert category, collected on a certain date.
+
+    Args:
+        category (str): An MSE category
+        collection_date (str): A date in the format "YYYY_MM_DD"
+    Returns:
+        pd.DataFramedataframe with most up to date category/sub-forum data
+    """
+    if category == "energy":
+        return load_s3_data(
+            bucket_name=S3_BUCKET,
+            file_path=f"data/mse/outputs/mse_data_category_energy.parquet",
+        )
+
+    return load_s3_data(
+        bucket_name=S3_BUCKET,
+        file_path=f"data/mse/outputs/mse_data_category_{category}_{collection_date}.parquet",
+    )
+
+
+def get_all_mse_data(collection_date: str) -> pd.DataFrame:
+    """
+    Gets data from all Money Saving Expert categories.
+
+    Args:
+        collection_date (str): A date in the format "YYYY_MM_DD"
+    Returns:
+        pd.DataFrame: a dataframe with data from all categories
+    """
+    categories = [
+        "green-ethical-moneysaving",
+        "lpg-heating-oil-solid-other-fuels",
+        "energy",
+        "is-this-quote-fair",
+    ]
+    all_mse_data = pd.DataFrame()
+    for cat in categories:
+        if (
+            cat != "energy"
+        ):  # this is a temporary fix, while we sort the collection for this category
+            aux = load_s3_data(
+                bucket_name=S3_BUCKET,
+                file_path=f"data/mse/outputs/mse_data_category_{cat}_{collection_date}.parquet",
+            )
+        else:
+            aux = load_s3_data(
+                bucket_name=S3_BUCKET,
+                file_path=f"data/mse/outputs/mse_data_category_energy.parquet",
+            )
+        all_mse_data = pd.concat([all_mse_data, aux])
+
+    return all_mse_data.drop_duplicates().reset_index(drop=True)
+
+
+def get_mse_data(category: str, collection_datetime: str) -> pd.DataFrame:
```

Comment:
I find the name (and doc strings) of these functions "get_all_mse_data" and "get_category_mse_data" a little bit confusing as they appear to have the same functionality (except this one uses those functions). Not sure if there is a description/way to make them more distinguishable like describing it as a wrapper/decorator (or something similar).